openshift/helper/__init__.py

```diff
@@ -209,178 +181,128 @@ def delete_object(self, name, namespace, delete_opts=None):
-        """
-        self.__log_argspec()
-        obj = self.model()
-        obj.kind = string_utils.snake_case_to_camel(self.kind).capitalize()
-        obj.api_version = self.api_version.lower()
-        for param_name, param_value in module_params.items():
-            if self.argspec[param_name].get('property_path'):
-                spec = self.argspec[param_name]
-                prop_path = copy.copy(spec['property_path'])
-                self.__set_obj_attribute(obj, prop_path, param_value)
-        logger.debug("Object from params:")
-        logger.debug(json.dumps(obj.to_dict(), indent=4))
-        return obj
-
-    def __set_obj_attribute(self, obj, property_path, param_value):
-        """
-        Recursively set object properties
-        :param obj: The object on which to set a property value.
-        :param property_path: A list of property names in the form of strings.
-        :param param_value: The value to set.
-        :return: The original object.
-        """
-        logger.debug("set {0}, {1} to {2}".format(obj.__class__.__name__,
-                                                  json.dumps(property_path),
-                                                  json.dumps(param_value)))
-        while len(property_path) > 0:
-            prop_name = property_path.pop(0)
-            prop_kind = obj.swagger_types[prop_name]
-            if prop_kind in ('str', 'int', 'bool') or \
-                    prop_kind.startswith('dict(') or \
-                    prop_kind.startswith('list['):
-                # prop_kind is a primitive
-                setattr(obj, prop_name, param_value)
+    def replace_object(self, name, namespace, k8s_obj, wait=False, timeout=60):
+        empty_status = self.properties['status']['class']()
+        k8s_obj.status = empty_status
+        self.__remove_creation_timestamps(k8s_obj)
+        try:
+            replace_method = self.__lookup_method('replace', namespace)
+            if namespace is None:
+                return_obj = replace_method(name, k8s_obj)
             else:
-                # prop_kind is an object class
-                sub_obj = getattr(obj, prop_name)
-                if not sub_obj:
-                    sub_obj = getattr(client.models, prop_kind)()
-                    logger.debug("sub_obj is {}".format(sub_obj.__class__.__name__))
-                logger.debug("set {0}.{1}".format(obj.__class__.__name__, prop_name))
-                setattr(obj, prop_name, self.__set_obj_attribute(sub_obj, property_path, param_value))
+                return_obj = replace_method(name, namespace, k8s_obj)
+        except ApiException as exc:
+            msg = json.loads(exc.body).get('message', exc.reason)
+            raise OpenShiftException(msg, status=exc.status)
+        if wait:
+            return_obj = self.__wait_for_response(k8s_obj.metadata.name, namespace, timeout)
+        return return_obj
+
+    def __wait_for_response(self, name, namespace, timeout, delete=False):
```

Comment:
Much better than the previous implementation, but it might be better to use a watch instead: https://github.com/kubernetes-incubator/client-python#example